pkg/vault/contracts/CompositeLiquidityRouterHooks.sol

```diff
@@ -128,9 +128,19 @@ contract CompositeLiquidityRouterHooks is BatchRouterCommon {
             unwrapWrapped.length
         );
 
-        (, uint256[] memory actualAmountsOut, ) = _vault.removeLiquidity(
-            _buildRemoveLiquidityParams(params, numTokens)
-        );
+        uint256[] memory actualAmountsOut;
+
+        // If the pool is in Recovery Mode, do a recovery withdrawal.
+        if (_vault.isPoolInRecoveryMode(params.pool)) {
+            actualAmountsOut = _vault.removeLiquidityRecovery(
+                params.pool,
+                params.sender,
+                params.maxBptAmountIn,
+                params.minAmountsOut
+            );
+        } else {
+            (, actualAmountsOut, ) = _vault.removeLiquidity(_buildRemoveLiquidityParams(params, numTokens));
+        }
```

Comment:
huh this is very interesting, I hadn't thought about this possibility. In my head I was thinking about a different entrypoint (i.e. a remove liquidity recovery external function) that passes a flag to the hook, and then based on that flag you'd call one of the options.

I really don't have a strong opinion; both options are valid. I had a brief chat with @danielmkm, and he's leaning towards two different entrypoints (same approach as the current regular router). The frontend already does the pivot when we're in recovery mode. Does it fit the bytecode if we do different entrypoints? Would the extra flag cause stack too deep issues?